endpoints/resources/src/main.cpp

```diff
@@ -221,49 +222,133 @@ namespace
 
 	IRODS_HTTP_API_ENDPOINT_OPERATION_SIGNATURE(op_modify)
 	{
-#if 0
         auto result = irods::http::resolve_client_identity(_req);
         if (result.response) {
             return _sess_ptr->send(std::move(*result.response));
         }
 
         const auto client_info = result.client_info;
 
-        irods::http::globals::background_task([fn = __func__, client_info, _sess_ptr, _req = std::move(_req), _args = std::move(_args)] {
-            logging::info("{}: client_info.username = [{}]", fn, client_info.username);
+		irods::http::globals::background_task([fn = __func__,
+		                                       client_info,
+		                                       _sess_ptr,
+		                                       _req = std::move(_req),
+		                                       _args = std::move(_args)] {
+			logging::info("{}: client_info.username = [{}]", fn, client_info.username);
+
+			http::response<http::string_body> res{http::status::ok, _req.version()};
+			res.set(http::field::server, irods::http::version::server_name);
+			res.set(http::field::content_type, "application/json");
+			res.keep_alive(_req.keep_alive());
+
+			try {
+				const auto name_iter = _args.find("name");
+				if (name_iter == std::end(_args)) {
+					logging::error("{}: Missing [name] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(http::status::bad_request));
+				}
+
+				const auto property_iter = _args.find("property");
+				if (property_iter == std::end(_args)) {
+					logging::error("{}: Missing [property] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(http::status::bad_request));
+				}
+
+				const auto value_iter = _args.find("value");
+				if (value_iter == std::end(_args)) {
+					logging::error("{}: Missing [value] parameter.", fn);
+					return _sess_ptr->send(irods::http::fail(http::status::bad_request));
+				}
+
+				auto conn = irods::get_connection(client_info.username);
+
+				if (property_iter->second == "name") {
+					// TODO(???): Remove this once the resource administration library grows support
+					// for renaming resources.
+					//
+					// The following is equivalent to what the resource administration library would
+					// provide.
+					GeneralAdminInput input{};
+					input.arg0 = "modify";
+					input.arg1 = "resource";
+					input.arg2 = name_iter->second.c_str();
+					input.arg3 = "name";
```

Comment:
At the risk of being less clear... this could be `property_iter->second.c_str()` instead of just "name".

Reply:
It could, but I don't see much value in doing that for this case. Using the string literal removes any reliance on the compiler optimizing away potential memory accesses.

This is also identical to what will land in the resource administration lib.

I'm going to leave this as is for now.